response_test.go

```diff
@@ -602,6 +602,22 @@ func TestResponse_BodyDeferred(t *testing.T) {
 		assert.Nil(t, resp.content)
 		assert.Equal(t, contentFailed, resp.contentState)
 	})
+
+	t.Run("content hijacked", func(t *testing.T) {
```

Comment:
In this test, we need some additional checks:

* that response is failed
* that returned String is failed (and IsEmpty is not actually legit, please remove it)
* that there were no attempt to read or close mock body
* that content state didn't change (still hijacked)